doc/hardware/cache/index.rst

```diff
@@ -1,9 +1,146 @@
 .. _cache-guide:
 
-Cache Interface
-###############
-
-This is a high-level guide to cache interface and Kconfig options related to
+Caching Basics
+##############
+
+This section discusses the basics of cache coherency and under what situations a user
+needs to explicitly deal with caching. For more detailed info on Zephyr's caching
+tools, see :ref:`_cache_config` for Zephyr Kconfig options or :ref:`cache_api` for
+the API reference. This section primarily focuses on the data cache though there is
+typically also an instruction cache for systems with cache support.
+
+.. note::
+
+  The information here assumes that the architecture-specific MPU support is enabled.
+  See the architecture-specific documentation for details.
+
+When dealing with multiple bus masters that do not share the same cache, the issue
+of cache coherency becomes an issue. This includes SMP processors or microcontrollers
+as well as multi-master systems like DMA controllers. Typically processor caches
+exist as close to each processor core as possible to maximize performance gain. Because
+of this, data moved into and out of memory by DMA engines or a different processor
+core will be stale in the processor's cache, resulting in what appears to be corrupt
+data. If you are moving data using DMA and the processor doesn't see the data you
+expect, cache coherency may be the issue.
+
+There are multiple approaches to ensuring that the data seen by the processor core
+and the peripherals is identical (cache coherency). The simplest is just to disable
+caching, but this defeats the purpose of having a hardware cache in the first place
+and results in a significant performance hit. Many architectures provide methods for
+disabling caching for only a portion of memory. This can be useful when cache coherence
+is more important than performance, such as when using DMA with SPI. Finally, there
+is the option to flush or invalidate the cache for regions of memory at runtime.
+
+Globally Disabling the Data Cache
+---------------------------------
+
+As mentioned above, globally disabling data caching can have a significant performance
+impact but can be useful for debugging.
+
+* Requirements:
+  * :kconfig:option:`CONFIG_DCACHE`: DCACHE control enabled in Zephyr.
+  * :kconfig:option:`CONFIG_CACHE_MANAGEMENT`: cache API enabled.
+  * Call :cpp:func:`sys_cache_data_disable()` to globally disable the data cache.
```

Comment:
```suggestion
  * Call :c:func:`sys_cache_data_disable()` to globally disable the data cache.
```

elsewhere too

Reply:
Thanks, I'm still a bit new to sphinx. I'm working through building these locally and fixing the errors...

Reply:
No worries, this one just caught my eye. Let me know if you are having issues setting up the local build environment and if there is anything I can do to help

Reply:
It took a bit to figure out how to get `sphinx-build` not to OOM my whole machine, but once that was solved I was able to fix the other errors. Hopefully the docs should now be reviewable in the CI build once it's complete.